tests/backup/backup_psa_test.go

```diff
@@ -943,3 +944,440 @@ var _ = Describe("{RestoreFromHigherPrivilegedNamespaceToLower}", Label(TestCase
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of restore with %s from backup %s", restoreName, baselineManualBackupName))
+		})
+
+		Step("Creating schedule backup with all restricted & baseline namespaces", func() {
+			log.InfoD("Creating schedule backup with all restricted & baseline namespaces")
+			schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(BackupOrgID, ctx, periodicPolicyName)
+			schBackupName := fmt.Sprintf("schbkp-all-ns-%v", RandomString(4))
+			scheduleNames = append(scheduleNames, schBackupName)
+			labelSelectors := make(map[string]string)
+			log.InfoD("Creating a schedule backup [%s] with namespaces [%s]", backupName, bkpNamespaces)
+			allNSScheduleBackup, err = CreateScheduleBackupWithValidation(ctx, schBackupName, SourceClusterName, backupLocationName, backupLocationUID, scheduledAppContexts, labelSelectors, BackupOrgID, preRuleName, preRuleUid, postRuleName, postRuleUid, periodicPolicyName, schPolicyUid)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of schedule backup [%s] with all namespaces contains both restricted and baseline apps.", allNSScheduleBackup))
+			schBackupList = append(schBackupList, allNSScheduleBackup)
+			log.InfoD("Schedule backup list after all namespaces backup %v", schBackupList)
+		})
+
+		wg.Add(1)
+		go func() {
+			defer GinkgoRecover()
+			defer wg.Done()
+			Step("Creating schedule backup with restricted namespaces", func() {
+				log.InfoD("Creating schedule backup with restricted namespaces")
+				schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(BackupOrgID, ctx, periodicPolicyName)
+				schBackupName := fmt.Sprintf("schbkp-restricted-%v", RandomString(5))
+				mutex.Lock()
+				scheduleNames = append(scheduleNames, schBackupName)
+				mutex.Unlock()
+				labelSelectors := make(map[string]string)
+				restrictedScheduleBackup, err = CreateScheduleBackupWithValidation(ctx, schBackupName, SourceClusterName, backupLocationName, backupLocationUID, restrictedScheduledAppContexts, labelSelectors, BackupOrgID, preRuleName, preRuleUid, postRuleName, postRuleUid, periodicPolicyName, schPolicyUid)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of schedule backup %v with restricted namespaces", restrictedScheduleBackup))
+				mutex.Lock()
+				schBackupList = append(schBackupList, restrictedScheduleBackup)
+				mutex.Unlock()
+				log.InfoD("Schedule backup list after schedule backup with restricted namespaces - %v", schBackupList)
+			})
+		}()
+
+		wg.Add(1)
+		go func() {
+			defer GinkgoRecover()
+			defer wg.Done()
+			Step("Creating schedule backup with baseline namespaces", func() {
+				log.InfoD("Creating schedule backup with baseline namespaces")
+				schPolicyUid, _ = Inst().Backup.GetSchedulePolicyUid(BackupOrgID, ctx, periodicPolicyName)
+				schBackupName := fmt.Sprintf("schbkp-baseline-%v", RandomString(5))
+				mutex.Lock()
+				scheduleNames = append(scheduleNames, schBackupName)
```

Comment:
Can we do it in a single mutex lock? Yes it will save only a few milliseconds but would be a better way to do.